File: 0962 Maximum Width Ramp/0962.cpp
class Solution {
public:
    int maxWidthRamp(vector<int>& nums) {
        stack<int> stk;
        
        for (int i = 0; i < nums.size(); i++){
            if (stk.empty() || nums[stk.top()] > nums[i]){
                stk.push(i);
            } 
        }

        int ans = 0;

        for (int i = nums.size() - 1; i >= 0; i--){
            while (!stk.empty() && nums[stk.top()] <= nums[i]){
                ans = max(ans, i - stk.top());
                stk.pop();
            }
        }

        return ans;
    }
};

File: 0962 Maximum Width Ramp/0962.py
class Solution:
    def maxWidthRamp(self, nums: List[int]) -> int:
        stk = []
        n = len(nums)

        for i in range(n):
            if not stk or nums[stk[-1]] > nums[i]:
                stk.append(i)

        ans = 0

        for i in range(n-1, -1, -1):
            while stk and nums[stk[-1]] <= nums[i]:
                ans = max(ans, i - stk.pop())
    
        return ans
        
